[MM-28483] Surface custom about modal for cloud instances

nickmisasi opened the pull request:
#### Summary
The default about modal reveals a lot of information that a Cloud user will not care about (and probably shouldn't see for security reasons). This PR adds a custom about modal that appears for Cloud licenses. It utilizes the default AboutCloudModal as an entry-point.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28483

#### Related Pull Requests
n/a

#### Screenshots

![image](https://user-images.githubusercontent.com/12176405/93127053-3b17ca00-f69b-11ea-843b-8d996b6138e3.png)

devinbinnie commented on components/about_build_modal/about_build_modal.jsx:
```diff
@@ -12,6 +12,8 @@ import Nbsp from 'components/html_entities/nbsp';
 
 import {AboutLinks} from 'utils/constants';
 
+import AboutBuildModalCloud from './about_build_modal_cloud/about_build_modal_cloud';
```
For future reference, if you're not connecting the component to redux, you don't need to create a subfolder for it.

devinbinnie commented on components/about_build_modal/about_build_modal.jsx:
```diff
@@ -59,6 +61,16 @@ export default class AboutBuildModal extends React.PureComponent {
         const license = this.props.license;
+        const isCloud = license.Cloud === 'true';
+
+        if (isCloud) {
```
Can we move the boolean statement here, just to reduce code bloat?

devinbinnie reviewed (approved):
LGTM, just a couple small comments.

matthewbirtch reviewed (changes requested):
Nice work @nickmisasi. Two small things I noticed:
- Can we change the padding on `.modal-body` to be:  `padding: 15px 64px`
- If I switch to a dark theme, the line "High trust messaging for the enterprise" is hard to see. It needs to use the proper theme color for the center channel text color
![image](https://user-images.githubusercontent.com/2040554/93131861-b630ae80-f6a2-11ea-81a7-e30d415b57e9.png)

Other than that, I think it's good to go!

marianunez commented on components/about_build_modal/about_build_modal_cloud/about_build_modal_cloud.scss:
```diff
@@ -0,0 +1,72 @@
+                top: 32px;
+                font-size: 16px;
+                line-height: 24px;
+                color: #3D3C40;
```
We should use the theme color here and may be what Matt was referring to in dark mode.

nickmisasi replied:
Yeah definitely the culprit

marianunez commented on components/about_build_modal/about_build_modal_cloud/about_build_modal_cloud.tsx:
```diff
@@ -0,0 +1,122 @@
+                id='about.licensed'
+                defaultMessage='Licensed to:'
+            />
+            {'\u00a0' + license.Company}
```
For company we want to show the company information when they signed up for cloud and not the company of the license because in this case is always Mattermost. 

We can add this later when we have the endpoints to get Company information from CWS which we will have for Billing in the System Console Soft GA. Created a ticket https://mattermost.atlassian.net/browse/MM-28698

marianunez reviewed (approved):
Looks great @nickmisasi! Non-blocking comment below

nickmisasi commented:
Fixed the text, added the padding as requested @matthewbirtch 

With that new padding it doesn't necessarily match the design, distance between the top of the mattermost svg and the header is 46 pixels in figma, but now with 15 px padding on top its 61. Let me know what you think.

![image](https://user-images.githubusercontent.com/12176405/93224623-4c1a1700-f73f-11ea-955d-e8ba0506c027.png)
![image](https://user-images.githubusercontent.com/12176405/93224640-550ae880-f73f-11ea-9f80-57b3c0506dbe.png)

matthewbirtch commented:
> but now with 15 px padding on top its 61. Let me know what you think.

Ya, you're right. With the 15px on top, it feels off balance. I think you can make that top padding 0 actually. That should do the trick

nickmisasi commented:
@matthewbirtch back to 0px padding at the top on the next spinwick update. Let me know what you think!

stevemudie reviewed (approved):
LGTM

matthewbirtch reviewed (approved):
Looks good @nickmisasi. Thanks!

nickmisasi commented:
/cherry-pick cloud-beta

mattermod commented:
Cherry pick is scheduled.
